internal/credit/postgres_connector/balance.go

```diff
@@ -193,6 +200,9 @@ func (a *PostgresConnector) getBalance(
 			if grantBalance.EffectiveAt.After(period.To) {
 				continue
 			}
+			if periodIndex != 0 && grantBalance.EffectiveAt.Equal(period.To) {
```

Comment:
If I understand correctly what you want to achieve here, we would only query the meter for the first feature IMHO if multiple grants are created in the same window period for multiple features. 

I would highly recommend adding a test case to multiple features with this grant setup.

Reply:
I'm not sure I understand what you mean. There's a testcase `Should include usage between ledger creation and first grant`. What this bit does is burns down the first grant with that usage. Thinking about it:
- IMO this makes sense because the user would want to create ledgers and grants together (when first issuing), but it's reasonable to expect some delay between those operations, either through Cloud UI, or via API integration in selfhosted
- If they don't want to acknowledge that usage (e.g. create the ledger preemptively and later start issuing grants) they can always do reset for the ledger, then usage is only burnt down from the reset.

Reply:
What I meant is if at time T we have a
- Reset
- Grant for feature1 in the same window as T is in
- Grant for feature2 in the same window as T is in

We wouldn't fetch the meter for one of the features.

Reply:
goccha, fixed it